packages/framework-provider-aws-infrastructure/src/infrastructure/stacks/permissions.ts

```diff
@@ -6,43 +6,51 @@ import { GraphQLStackMembers } from './graphql-stack'
 import { ScheduledCommandStackMembers } from './scheduled-commands-stack'
 import { EventsStackMembers } from './events-stack'
 import { Function } from '@aws-cdk/aws-lambda'
+import { BoosterConfig } from '@boostercloud/framework-types'
 
 export const setupPermissions = (
+  config: BoosterConfig,
   graphQLStack: GraphQLStackMembers,
   eventsStack: EventsStackMembers,
   readModelTables: Array<Table>,
-  websocketAPI: CfnApi,
+  websocketAPI?: CfnApi,
   scheduledCommandStack?: ScheduledCommandStackMembers
 ): void => {
   const { graphQLLambda, subscriptionsStore, subscriptionNotifier, connectionsStore } = graphQLStack
   const { eventsLambda, eventsStore } = eventsStack
   const scheduledLambda = scheduledCommandStack?.scheduledLambda
-  const websocketManageConnectionsPolicy = createPolicyStatement(
-    [
-      Fn.join(':', [
-        'arn',
-        Fn.ref('AWS::Partition'),
-        'execute-api',
-        Fn.ref('AWS::Region'),
-        Fn.ref('AWS::AccountId'),
-        `${websocketAPI.ref}/*`,
-      ]),
-    ],
-    ['execute-api:ManageConnections']
-  )
+  if (config.enableSubscriptions && websocketAPI) {
+    const websocketManageConnectionsPolicy = createPolicyStatement(
+      [
+        Fn.join(':', [
+          'arn',
+          Fn.ref('AWS::Partition'),
+          'execute-api',
+          Fn.ref('AWS::Region'),
+          Fn.ref('AWS::AccountId'),
+          `${websocketAPI.ref}/*`,
+        ]),
+      ],
+      ['execute-api:ManageConnections']
+    )
+    if (subscriptionsStore && connectionsStore && subscriptionNotifier) {
+      grantFullAccessToSubscriptionsStore(subscriptionsStore, graphQLLambda)
+      graphQLLambda.addToRolePolicy(
+        createPolicyStatement(
+          [connectionsStore.tableArn],
+          ['dynamodb:GetItem', 'dynamodb:PutItem', 'dynamodb:DeleteItem']
+        )
+      )
+      graphQLLambda.addToRolePolicy(websocketManageConnectionsPolicy)
+      // Subscription notifier lambda permissions
+      subscriptionNotifier.addToRolePolicy(createPolicyStatement([subscriptionsStore.tableArn], ['dynamodb:Query*']))
+      subscriptionNotifier.addToRolePolicy(websocketManageConnectionsPolicy)
+    }
+  }
```

Comment:
I think we can simplify the `ifs` by not asking for `config.enableSubscriptions` again, as, if it is false, all the other members related to subscriptions are undefined (the reason is to avoid duplicity in the logic to know whether the subscriptions are enabled or not). Something like:
```suggestion
  if (websocketAPI) {
    const websocketManageConnectionsPolicy = createPolicyStatement(
      [
        Fn.join(':', [
          'arn',
          Fn.ref('AWS::Partition'),
          'execute-api',
          Fn.ref('AWS::Region'),
          Fn.ref('AWS::AccountId'),
          `${websocketAPI.ref}/*`,
        ]),
      ],
      ['execute-api:ManageConnections']
    )
  }
  if (subscriptionsStore && connectionsStore && subscriptionNotifier) {
    grantFullAccessToSubscriptionsStore(subscriptionsStore, graphQLLambda)
    graphQLLambda.addToRolePolicy(
      createPolicyStatement(
        [connectionsStore.tableArn],
        ['dynamodb:GetItem', 'dynamodb:PutItem', 'dynamodb:DeleteItem']
      )
    )
    graphQLLambda.addToRolePolicy(websocketManageConnectionsPolicy)
    // Subscription notifier lambda permissions
    subscriptionNotifier.addToRolePolicy(createPolicyStatement([subscriptionsStore.tableArn], ['dynamodb:Query*']))
    subscriptionNotifier.addToRolePolicy(websocketManageConnectionsPolicy)
  }
}
```

Reply:
See next comment for what I believe is a better approach